pkg/telemetry/mocks/mock_tracing_server.go

```diff
@@ -0,0 +1,48 @@
+package mocks
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"sync"
+
+	otlpcollector "go.opentelemetry.io/proto/otlp/collector/trace/v1"
+	"google.golang.org/grpc"
+)
+
+type MockTracingServer struct {
+	otlpcollector.UnimplementedTraceServiceServer
+	exportCount int
+	serviceMu   sync.Mutex
+}
+
+func (s *MockTracingServer) Export(context.Context, *otlpcollector.ExportTraceServiceRequest) (*otlpcollector.ExportTraceServiceResponse, error) {
+	s.serviceMu.Lock()
+	defer s.serviceMu.Unlock()
+	s.exportCount++
+	return &otlpcollector.ExportTraceServiceResponse{}, nil
+}
+
+func NewMockTracingServer(port int) (*MockTracingServer, error) {
```

Comment:
```suggestion
func NewMockTracingServer(port int) (otlpcollector.TraceServiceServer, error) {
   	mockServer := &mockTracingServer{exportCount: 0}

	go func() {
		lis, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
		if err != nil {
			log.Fatalf("failed to listen: %v", err)
		}
		server := grpc.NewServer()
		otlpcollector.RegisterTraceServiceServer(server, mockServer)
		if err := server.Serve(lis); err != nil {
			log.Fatalf("failed to serve: %v", err)
		}
	}()
	
	return mockServer, nil
}
```

Reply:
I found this article https://medium.com/@cep21/preemptive-interface-anti-pattern-in-go-54c18ac0668a that says that it's better to return structs.

Reply:
The guidance of "Accept interfaces, return structs" is good general advice, but not always applicable. In this case, the only thing we're really testing for is that the function `Export` of the mocked interface (`TraceServiceServer`) is called exactly once. So creating a dependency on a custom struct when the actual unit under tests is a mocked interface seems counterproductive here. This issue goes away though if we can find a good approach for https://github.com/openfga/openfga/pull/822#discussion_r1244456438

Let me get back with ya.

Reply:
So... I added a method `GetExportCount` in the mock that is used in the test to assert that  the `Export` method got called. If I change the return type, I lose access to this method.

Unless I figure out how to use `mockgen`, I don't see a way of solving this elegantly :/